low-level/framework/utils/salt_util.py

```diff
@@ -13,109 +13,171 @@
-                    logger.warning("Using node_id ('srvnode-1') as we are not able to fetch it from hostname command.")
-                    node_key = 'srvnode-1'
-                else:
-                    node_key = result[0].decode().rstrip('\n').lower()
+            with open(CommonConfig.SALT_FILE, 'r') as salt_file:
+                _node_id = salt_file.read().rstrip()
+            if _node_id is None:
+                _node_id = 'srvnode-1'
+            logger.info(f'salt_util, Server minion ID: {_node_id}')
+        except OSError as err:
+            if err.errno == errno.ENOENT:
+                logger.error(
+                    f'Problem occured while reading from salt file. \
+                     File path: {CommonConfig.SALT_FILE} does not exist')
+            elif err == errno.EACCES:
+                logger.error(
+                    f'Problem occured while reading from salt file. \
+                     Permission issue while reading from: {CommonConfig.SALT_FILE}')
             else:
-                node_key = result[0].decode().rstrip('\n')
-        except Exception as e:
-            logger.warning(f"Can't read node id, using 'srvnode-1' : {e}")
-        return node_key
+                logger.error(
+                    f'Problem occured while reading from salt file with \
+                     the issue: {err}')
+        except Exception as err:
+            logger.warning(f'salt_util, Fail to read node-id with error: {err}, \
+                             Keeping default value as srvnode-1')
+        return _node_id
 
-    def is_single_node(self):
+    def _is_server_single(self, _err=None):
         """
         Returns true if single node, false otherwise
         """
-        is_single_node = True
+        _is_single_node = False
+
         try:
-            SALT_CMD = "sudo salt-call pillar.get cluster:type --output=newline_values_only"
-            subout = subprocess.Popen(SALT_CMD, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-            result = subout.stdout.readlines()
-            if result == [] or result == "":
-                logger.warning("Cluster type fetch failed, assuming single node setup.")
+            if self.pillar_info:
+                cluster_type = \
+                    self.pillar_info[self.node_id][self.CLUSTER_KEY][self.CLUSTER_TYPE_KEY]
+                if cluster_type is not None and cluster_type.lower() == 'single':
+                    _is_single_node = True
+                logger.info(f'salt_util, Node type: {_is_single_node}')
             else:
-                if result[0].decode().rstrip('\n') != "single":
-                    is_single_node = False
-        except Exception as e:
-            logger.warning(f"Cluster type read failed, assuming single node setup : {e}")
-        return is_single_node
+                logger.warning(f'salt_util, Fail to read cluster type with \
```

Comment:
in else or exception case, '_is_single_node' is not set to 'True'.

Reply:
So, my implementation is:
Default I am keeping it as False. If we get 'single' from Prov API. then it will be set to True. So, In case of error and exception, it will use default which is already set at line number 100.

Reply:
Ok